### DISCUSSION TITLE
Can I obtain the Fritzbox' password on the back of the router from CLI

### DISCUSSION THREAD

**Discussion by fraternl** [Freetz-NG/freetz-ng#875] [Allgemein] [answered]
I would like to give the Freetz interface the same password as is written on the back of the modern.

Anyone already knows how to get it?

![image](https://github.com/Freetz-NG/freetz-ng/assets/8692856/b1bc1622-9e44-46ef-af34-646b648ffd65)

**Comment by fraternl** [comment, marked as answer]
grep webgui_pass  /proc/sys/urlader/environment
webgui_pass     cycles1526

**Comment by fraternl** [comment]
After a factory reset the password for the Freetz interface becomes "freetz" so does the password for root login from the terminal.

I would like to give the box the password that's written on the back of the box.
Can someone tell me the most elegant way of changing the freetz webif password?


PSEUDOCODE
```

BACKPASS=`grep '^webgui_pass'  /proc/sys/urlader/environment | awk '{print $2}' | egrep -o '[a-z]+[0-9]+'`
# Make sure I have a valid 10 character password
if echo -n "${BACKPASS}" | wc -c | grep '^10$' ; then

   if <FreetzWebiIFpassword is "freetz"
      change FreetzWebiIFpassword to  "${BACKPASS}"
  fi
fi

```

I would also like to know how to find out that the password for root is still "freetz"
After a factory reset one needs to change the password from _freetz_ to another.
The script that's called should give me some clues

**Comment by fda77** [comment]
Check the hash in /etc/passwd and replace it if its the freetz factory value.

**Comment by fraternl** [comment]
I ended up with this code which I have in a provision script which only executes if the server is in factory default.
If the variable REBOOT_REQUIRED is set it will also do a _**modsave all**_ later on in the script before it reboots (with a delay)

I also needed to enable the busybox applet **_chpasswd_**

```
BACKPASS=`grep '^webgui_pass'  /proc/sys/urlader/environment | awk '{print $2}' | egrep -o '[a-z]+[0-9]+'`
if echo -n "${BACKPASS}" | wc -c | grep -q '^10$' ; then
  # If root passwd is "freetz"
  if grep -q "^root:\$1\$\$zO6d3zi9DefdWLMB" /etc/shadow ; then
    echo "Change root passwd from freetz to ${BACKPASS}"  >>${LOG}
    echo "root:${BACKPASS}" | chpasswd
    REBOOT_REQUIRED=true
  fi
fi

```

Both methods work for changing the password.
There's no need to add the applet **_chpasswd_**


```
    echo "root:${BACKPASS}" | chpasswd
    echo -e "${BACKPASS}\n${BACKPASS}" | passwd

```

**Comment by fda77** [comment]
You could use `modconf` to set a variable. Imho there should be in the old Wiki some notes about it
